stratum/hal/lib/phal/phal.cc

```diff
@@ -59,22 +57,24 @@ ::util::Status Phal::PushChassisConfig(const ChassisConfig& config) {
   absl::WriterMutexLock l(&config_lock_);
 
   if (!initialized_) {
-    // Do init stuff here
+    // Create attribute DB.
     std::unique_ptr<AttributeGroup> root_group =
         AttributeGroup::From(PhalDB::descriptor());
     std::vector<std::unique_ptr<SwitchConfiguratorInterface>> configurators;
 
-    // Set up ONLP
-#if defined(WITH_ONLP)
-    {
+    // Set up ONLP plugin.
+    if (FLAGS_enable_onlp) {
       auto* onlp_wrapper = onlp::OnlpWrapper::CreateSingleton();
+      CHECK_RETURN_IF_FALSE(onlp_wrapper != nullptr)
```

Comment:
if nullptr, then message and continue

Reply:
With the runtime flag (`enable_onlp`) we can prevent this kind of grey failure, where ONLP is requested (either implicitly by default, or explicitly) but then is not available.

In case of ONLP errors, the user would have to consciously disable ONLP. Same decision as previously, now st runtime instead of at compile time.

Reply:
Makes sense